packages/core/src/rules/common/spec.ts

```diff
@@ -35,6 +37,18 @@ export const OasSpec: Oas3Rule | Oas2Rule = () => {
         }
       }
 
+      const allowed = type.allowed && type.allowed(node);
+      if (allowed && isPlainObject(node)) {
+        for (const propName in node) {
+          if (!allowed.includes(propName)) {
+            report({
+              message: `The field \`${propName}\` is not allowed on this level.`,
```

Comment:
What if it isn't allowed at any level? I think we can simplify this a little.
```suggestion
              message: `The field \`${propName}\` is not allowed here.`,
```

Reply:
done